### PR TITLE
feat: add app.exception decorator to handle errors

### PR DESCRIPTION
**Pull request by IdoKendo**
**Description**

This PR addresses #480 

This PR adds the optional `app.exception` decorator under the Robyn app in order to allow catching exceptions raised in the routes and handle them gracefully.

Example usage added in `base_routes.py`

```python
@app.exception
def handle_exception(error):
    return {"status_code": 500, "body": f"error msg: {error}"}


@app.get("/sync/exception")
def sync_exception():
    raise ValueError("value error")
```

image

### CODE REVIEW

**Comment by sansyrox** [comment]
Hey @IdoKendo 👋 

Great work on the PR 😄 

Can you please add some integration tests in the `integrations_tests` folder?

**Comment by IdoKendo** [comment]
> Can you please add some integration tests in the `integrations_tests` folder?

Sure thing, added it. 😎

File: integration_tests/base_routes.py [hunk 1]
```diff
@@ -586,6 +586,19 @@ async def post(request: Request):
...
+
+
+@app.exception
+def handle_exception(error):
+    return {"status_code": 500, "body": f"error msg: {error}"}
+
+
+@app.get("/sync/exception")
+def sync_exception():
+    raise ValueError("value error")
```
Comment (sansyrox) [inline integration_tests/base_routes.py:599]: Hey @IdoKendo , 

Just two more suggestions - can we also test these exceptions for async code?

And can we add some test for methods with request bodies like POST, PUT ,etc?

Comment (IdoKendo) [reply integration_tests/base_routes.py:599]: Sure, will add them shortly!

Comment (IdoKendo) [reply integration_tests/base_routes.py]: @sansyrox I Added tests for async and for put and post, let me know if you think any more tests are needed.

**Comment by sansyrox** [comment]
Hey @IdoKendo 👋 

All looks good to me. 😄 Just there's one final merge conflict

**Comment by IdoKendo** [comment]
> Hey @IdoKendo 👋
> 
> All looks good to me. 😄 Just there's one final merge conflict

Hey @sansyrox no problem, I resolved the conflict! 😄

**Review by sansyrox — APPROVED** [review]
LGTM! Great work @IdoKendo 🔥

**Pull request merged**